### DISCUSSION TITLE
[Feature] API de de Corretoras em funcionamento no Brasil

### DISCUSSION THREAD

**Discussion by lucasbastianik** [BrasilAPI/BrasilAPI#213] [Ideas]
## Descrição
Acredito que seria legal, seguindo a linha de termos os dados de bancos, também as informações de corretoras que estão em funcionamento no Brasil.

### Origem dos dados
Para a origem dos dados, pensei em capturar as informações da CVM (Comissão Mobiliária de Valores) na qual possui uma lista de Intermediários onde possui dados cadastrais de de Participantes Intermediários como: Bancos Comerciais, Bancos de Investimentos, Bancos Múltiplos, Caixas Econômicas, Cooperativas de Crédito, Corretoras, Custodiantes, Distribuidoras, Entidades Administradoras de Mercado e Escrituradores.

Apesar de ver valor em criarmos outras rotas a partir das informações deste grupo de Intermediários e/ou talvez até cruzar com as informações dos Bancos, acredito que podemos partir das informações de corretoras e eventualmente expandir para outros desenvolvimentos.

A CVM disponibiliza [neste link](https://dados.gov.br/dataset/intermediario-cad) as seguintes informações:
- [Dicionário de dados (TXT)](http://dados.cvm.gov.br/dados/INTERMED/CAD/META/meta_cad_intermed.zip)
- [Dados de Intermediários (CSV)](http://dados.cvm.gov.br/dados/INTERMED/CAD/DADOS/cad_intermed.zip)

### Solução
Pensei em capturarmos os dados do CSV, mapearmos com base no dicionário em TXT e usarmos a mesma idéia de cache usada em `/api/banks/v1/**` para manter as informações por 24h diretamente na ponta do CDN. ⚡️

Existem 27 campos no dicionário de dados atualmente, mas pensei que o retorno poderia ser algo mais light como:
```json
{
   "cnpj":"02332886000104",
   "type":"CORRETORAS",
   "socialName":"XP INVESTIMENTOS CCTVM S.A.",
   "commercialName":"XP INVESTIMENTOS",
   "status":"EM FUNCIONAMENTO NORMAL"
}
```

### Rotas
- `GET /api/exchanges/v1/:cnpj`
- `GET /api/exchanges/v1`

Por fim mas não menos importante, gostaria de destacar o https://dados.gov.br/organization. Aqui tem uma série de informações que podem ser relevantes para desenvolvimento de API's futuras. Vale a pena dar uma olhada! 🤓

**Comment by RodriAndreotti** [comment]
Cara, esse endpoint ficou serviço muito massa!
Acho que complementar a isso, para quem utilizará o API para essa área de investimentos, daria até para pensarmos em criar  algo que verifique a situação de determinada instituição financeira junto ao FGC, o que acha?
No momento estou meio corrido, mas num futuro posso ajudar, se interessar.

Parabéns pela ideia e pelo PR ;-)
